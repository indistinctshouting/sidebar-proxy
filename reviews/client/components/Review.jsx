import React from 'react';
import TextBlock from './TextBlock.jsx';
import UserInfo from './UserInfo.jsx';
import ActionLinks from './ActionLinks.jsx';
import VoteButtons from './VoteButtons.jsx';
import ReviewHeader from './ReviewHeader.jsx';
import styled from 'styled-components';

const ReviewWrapper = styled.div`
  display:grid;
  grid-template-columns: 1fr 2fr;
  min-height: 240px;
  padding-top: 18px;
  padding-bottom: 17px;
  border-bottom: 1px solid #e6e6e6;
  :hover .action-links {
    display: grid;
    grid-template-rows: repeat(5, 19px);
    width: 217;
    height: 130px;
  }
`;

// get a better idea of the ratios for the text wrapper
const ReviewTextWrapper = styled.div`
  display: grid;
  grid-template-rows: min-content min-content 60px;
  grid-row-gap: 15px;
  padding: 0 15px;
`;

const ActionLinksWrapper = styled.div`
  display: none;
`;

const Review = ({review}) => {
  
  const formatName = (name) => {
    let names = name.split(' ');
    let lastInitial = `${names[1][0]}.`;
    return `${names[0]} ${lastInitial}`;
  };

  review.owner.name = formatName(review.owner.name);

  return (
    <ReviewWrapper>
      <div className="sidebar-wrapper">
        <UserInfo owner={review.owner}/>
        <ActionLinksWrapper className="action-links">
          <ActionLinks name={review.owner.name}/>
        </ActionLinksWrapper>
      </div>
      <ReviewTextWrapper>
        <div>
          <ReviewHeader 
            rating={review.rating} 
            date={review.date} 
            updated={review.updated}
            checkIns={review.owner.checkIns}
          />
        </div>
        <div><TextBlock className="text-block" text={review.text} /></div>
        <div className="vote-buttons"><VoteButtons votes={review.upvotes}/></div>
      </ReviewTextWrapper>
    </ReviewWrapper>
  );
};

export default Review;