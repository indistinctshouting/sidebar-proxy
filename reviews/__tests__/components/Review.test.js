import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Review from '../../client/components/Review';

const fakeReviews = [{ 
  owner: { 
    picture:
     'https://s3.amazonaws.com/uifaces/faces/twitter/dshster/128.jpg',
    name: 'Giovanny Reinger ',
    location: 'Caliton, VT',
    friends: 22,
    reviewCount: 112,
    photos: 37,
    checkIns: 8,
    elite: false 
  },
  upvotes: { useful: 10, funny: 2, cool: 8 },
  _id: '5b92c45cadf6b93d40b1ba6e',
  reviewId: 278,
  restaurantId: 38,
  rating: 3,
  date: '2013-08-13T08:51:59.518Z',
  text:
    'Dolores quo quisquam est cumque corrupti omnis voluptatem. Dolorum reiciendis molestiae sunt velit omnis nihil dolorum atque. Illum suscipit earum excepturi eius harum. Aut nihil tenetur cupiditate aut eos aut. Culpa eveniet dolor delectus natus.',
  updated: false,
  __v: 0 
}];

describe('<Review />', () => {
  describe('render()', () => {
    test('renders the component', () => {
      const wrapper = shallow(<Review review={fakeReviews}/>);

      expect(toJson(wrapper)).toMatchSnapshot();
    });
  });
});