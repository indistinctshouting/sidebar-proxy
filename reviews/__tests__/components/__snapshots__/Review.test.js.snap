// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Review /> render() renders the component 1`] = `
<div
  className="review-container"
>
  <div>
    <UserInfo />
  </div>
  <div
    className="review-wrapper"
  >
    <div
      className="rating-box"
    >
      <RatingBox />
    </div>
    <div>
      <TextBlock
        className="text-block"
      />
    </div>
  </div>
</div>
`;
